In [3]:
import pandas as pd

In [5]:
# from github
df_sms = pd.read_csv('https://raw.githubusercontent.com/Enrique1987/data/master/02_ML/03_spam.csv',sep=",", encoding='cp1252', usecols=["v1", "v2"])
df_sms = df_emails.rename(columns={'v1': 'label', 'v2': 'text'})
df_sms.head()

label                                               text
0   ham  Go until jurong point, crazy.. Available only ...
1   ham                      Ok lar... Joking wif u oni...
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...
3   ham  U dun say so early hor... U c already then say...
4   ham  Nah I don't think he goes to usf, he lives aro...

In [6]:
df_sms.describe()

label                    text
count   5572                    5572
unique     2                    5169
top      ham  Sorry, I'll call later
freq    4825                      30

What does it mean that from unique in text are just 5169 from 5572, lets see what are the duplicate.

In [38]:
df_duplicates = df_sms[df_sms.duplicated(['text'], keep='first')]
df_duplicates.count()

label    403
text     403
dtype: int64

In [35]:
df = pd.DataFrame({"A":["foo", "foo", "foo", "foo"], "B":[0,1,1,1], "C":["A","A","B","A"]})
df.head()

A  B  C
0  foo  0  A
1  foo  1  A
2  foo  1  B
3  foo  1  A

In [37]:
df2 = df.drop_duplicates(subset=['A', 'C'], keep='False')
df2.head()

A  B  C
2  foo  1  B

In [24]:
df_sms_count["text"]

count
label       
ham     4825
spam     747

In [18]:
df_sms_count.columns

MultiIndex([('text', 'count')],
           )